### ISSUE TITLE
404 error

### ISSUE THREAD

**Issue by minobedin** [eyecatchup/php-webmaster-tools-downloads#22]
I have some problem with the script. Last week it worked perfect for me but now the login generate this response and i don't understand why

Array ( [url] => https://www.google.com/accounts/ClientLogin
 [content_type] => text/plain
 [http_code] => 404
 [header_size] => 338
 [request_size] => 203
 [filetime] => -1
[ssl_verify_result] => 0
 [redirect_count] => 0
 [total_time] => 1.116964
 [namelookup_time] => 4.8E-5
 [connect_time] => 0.019938
 [pretransfer_time] => 0.074707
 [size_upload] => 609
 [size_download] => 65
 [speed_download] => 58
 [speed_upload] => 545
 [download_content_length] => 65
 [upload_content_length] => 609
 [starttransfer_time] => 1.075156
 [redirect_time] => 0
 [certinfo] => Array ( )
 [primary_ip] => 216.58.192.4
 [primary_port] => 443
 [local_ip] => 184.168.200.182
 [local_port] => 36531
 [redirect_url] => )

Anyone can help me?

**Comment by sblendorio** [comment]
As result of HTTP call we also received "https://developers.google.com/accounts/docs/AuthForInstalledApps" as output associated with a 404 error.

This page explicitly says that login method has been definitely migrated from ClientLogin to OAuth.
https://developers.google.com/accounts/docs/AuthForInstalledApps

Now I'm trying to find out how.

**Comment by sblendorio** [comment]
OAuth guide:
https://developers.google.com/identity/protocols/OAuth2

**Comment by leonrenkema** [comment]
Bumped into the same problem. It looks like the whole CSV download function is not available at the moment. Also with oauth2 token I cannot access the URL because of a mismatch in the URL. 
The new WMT api is using googleapis.com and this clientlogin is using google.com/webmasters.

**Comment by bacheson** [comment]
Changing the source name fixed the problem. Changing it from 'Google-WMTdownloadscript-0.1-php' to 'Google-WMTdownloadscript-0.1' miraculously started working. So Google is specifically blocking this library?

**Comment by leonrenkema** [comment]
Yup, just found that out :) That was also the fix for me.

**Comment by netcap-agency** [comment]
@bacheson Thanks, works great. Let's hope this 'library ban' is for an upcoming api update.

**Comment by minobedin** [comment]
thanks @bacheson, this works for me

**Comment by sblendorio** [comment]
This is a PHP project by Google for authentication through OAuth2:

https://github.com/google/google-api-php-client

**Comment by cchenship** [comment]
I did try @bacheson 's solution by change source to 'Google-WMTdownloadscript-0.1' at line 155 in gwtdata.php. But it did not work for me.
It did not produced any warnings or errors when I run the first example DownLoadCSV.php. 
I was able to identify that the LogIn did not go through.

I also thought ClientLogin was deprecated, but it appears that some of you are still using it OK? @minobedin  or @netcap-agency, did you change any other part of the code?

**Comment by netcap-agency** [comment]
@cchenship No, that is the only part that was changed. Maybe Google add an other filter to prevent the script to work. Other scripts using alternatives methods stop working again today. I have just checked to see if mine were affected but they are still working at this moment.

**Comment by cchenship** [comment]
Thanks, @netcap-agency , I probably came to the wrong post. I figured that my error is 403 (#21 issue) rather than 404 (#22 issue, i.e., the current post).

**Comment by minobedin** [comment]
@sblendorio I know that we can't use OAuth2 for download something like TOP_QUERIES. In my project i need it. With @bacheson solution it works now

**Comment by heathergaye** [comment]
@bacheson It was three days before I fly out of the country for a 7-week trip that our app stopped working, and you fixed it. Sobbing grovelling thanks from my entire company.

Does anyone have any idea of why this worked, though? Google's stance is that they've have disabled all access _except_ for their Python webmaster tools download library (with no comment on analogous third-party libraries, which makes sense). 

I assume source filtering is part of that mechanism, but I'd like to know whether it's to do with the source value format, or if they're specifically blocking certain values. If it's the latter, I expect this issue may happen again.

**Comment by bacheson** [comment]
not a problem @heathergaye :)

I talked to John Mueller (a higher up at GWT -> Google Search Console). They have every intention of keeping this feed live until the search console API is publicly available.

**Comment by tuscanicz** [comment]
@bacheson thank you much for tip, it's working like a magic. how did you know that?

**Comment by paulmoto** [comment]
@bacheson Thanks! I never would have thought to do that.

**Comment by 2alex2** [comment]
You have to use new Search Console API:

https://www.ukietech.com/blog/marketing/download-google-search-queries-using-google-search-console-api/

**Comment by etoseo** [comment]
Again this problem . Authorization fails

**Comment by leonrenkema** [comment]
I think this method is not working anymore, the new search console API is live and you should migrate to that.

**Comment by btibor91** [comment]
See also https://github.com/eyecatchup/php-webmaster-tools-downloads/issues/19#issuecomment-149804321.
